aws/resource_aws_ssm_maintenance_window_task.go

```diff
@@ -73,6 +73,18 @@ func resourceAwsSsmMaintenanceWindowTask() *schema.Resource {
 				},
 			},
 
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
```

Comment:
As per [the related documentation](https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_RegisterTaskWithMaintenanceWindow.html#systemsmanager-RegisterTaskWithMaintenanceWindow-request-Name), this is missing the Validation Constraints which are:

> Length Constraints: Minimum length of 3. Maximum length of 128.
> Pattern: ^[a-zA-Z0-9_\-.]{3,128}$

Reply:
Created the validator for this